File: CF_Quick-One/890_div2/A.py
from sys import stdin
def input(): return stdin.readline()[:-1]

def solve():
    N = int(input())
    A = list(map(int, input().split()))
    ans = 0
    for i in range(0, N-1):
        if A[i] > A[i + 1]:
            ans = max(ans, max(A[i], A[i + 1]))
    print(ans)

for _ in range(int(input())):
    solve()

File: CF_Quick-One/890_div2/A.java

// import java.util.Arrays;
import java.util.Scanner;

public class soln {

  private static final Scanner scanner = new Scanner(System.in);

  public static void main(String[] args) {

    int t = scanner.nextInt();
    while (t-- > 0) {
      solve();
    }
    scanner.close();

  }

  public static void solve() {
    int N = scanner.nextInt();
    int A[] = new int[N];
    for (int i = 0; i < N; i++) {
      A[i] = scanner.nextInt();
    }
    int ans = 0;
    for (int i = 0; i < N - 1; i++) {
      if (A[i] > A[i + 1]) {
        ans = Math.max(ans, Math.max(A[i], A[i + 1]));
      }
    }
    System.out.println(ans);
  }
}
